packages/autorest.go/src/transform/transform.ts

```diff
@@ -199,6 +199,7 @@ function substitueDiscriminator(item: Property | Parameter | SchemaResponse): Sc
       if (!discriminatorSchemas.has(sliceDiscriminators)) {
         discriminatorSchema = new ArraySchema(sliceDiscriminators, 'slice of discriminators', elementSchema);
         discriminatorSchema.language.go! = discriminatorSchema.language.default;
+        discriminatorSchema.language.go!.elementIsPtr = false;
```

Comment:
We are correctly generating this already, it just makes the metadata match the reality.